src/framework/uicomponents/view/sortfilterproxymodel.cpp

```diff
@@ -78,6 +78,7 @@ SortFilterProxyModel::SortFilterProxyModel(QObject* parent)
 
     connect(this, &SortFilterProxyModel::sourceModelRoleNamesChanged, this, [this]() {
         invalidate();
+        fillRoleIds();
```

Comment:
To add some explanation for @Eism: it turned out that `fillRoleIds()` was called before the source model was set, and thus `m_roleIdToFilterValueHash` stayed empty and the filter model did not do any filtering. Solution: call it again when the source model changes or when its role names change.